Lra 37 create an archive method to replace the delete destroy functionality

britaumich opened the pull request:
In addition to creating an archive method to replace the destroy one, I added 
- main tables' links to the header,
- authorize policies for the archive action, 
- required before_action authentication in controllers,
and reorganized the active admin navigation menu.

britaumich committed:
Add deleted_at field to main db tables

britaumich committed:
Add archive(d) scope and methods to models

britaumich committed:
Add temp tables links for easier navigation

britaumich committed:
Add policies for archive action

britaumich committed:
Add routes for archive method

britaumich committed:
Add archive method to controllers and archive action to views

britaumich committed:
Edit admin menus

britaumich committed:
Add active and archived scopes to active admin

britaumich committed:
Remove destroy action from views

rsmoke reviewed (changes requested):
Overall this looks great. i added in line comments requesting a conditional be set for the controllers that you give feedback after the archive method succeeds.

The header is an area that Maria can really provide guidance. I also would like to hide the specific layouts until a user is authenticated.

britaumich committed:
remove the unused/commented code

britaumich committed:
Hide links until a user authenticates

britaumich committed:
Add required authentication

britaumich committed:
Add check for unsuccessful archive; replace admin with admin_user

britaumich commented:
I removed commented/unused code;
hided navigation links until a user or admin_user authenticates;
replaced admin with admin_user in devise_group.
I added a check for the unsuccessful archive, but I don't know how to test it. Is that code correct?
I'll create a Jira ticket for Maria when navigation links are merged to master.

rsmoke reviewed (approved):
Great!
